### PR TITLE
DG update: add student implementation

### PR DESCRIPTION
**Pull request by chwoozy**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -133,6 +133,31 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+
+This feature is facilitated by `ListStudentCommand` which extends `Command`.
+The method `ListStudentCommand#execute` updates the filtered student list by calling the method
+`Model#updateFilteredStudentList` exposed in the `Model` interface.
+
+Given below is an example of how the list student 
+1. The user executes the list student command with the input `list_student`.
+2. `LogicManager` executes the input and parses the command using `AddressBookParser`.
+3. `AddressBookParser` identifies the correct command and creates a new `ListStudentCommand`.
+4. `AddressBookParser` returns the new `ListStudentCommand` to  `LogicManager`.
```
Comment (enhao25) [inline docs/DeveloperGuide.md]: I think there is additional space on to  `LogicManager`, but small problem ah.

File: docs/diagrams/choonwei/ListStudentSequenceDiagram.puml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+@startuml
+!include style.puml
```
Comment (enhao25) [inline docs/diagrams/choonwei/ListStudentSequenceDiagram.puml:2]: Actually you could include ../style.puml instead, so you don't have to make a new copy of the style.puml in your folder ah.

**Review by enhao25 — APPROVED** [review]
LGTM, just some minor comments ah.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (chwoozy) [reply docs/DeveloperGuide.md]: Thanks for spotting, will update.

File: docs/diagrams/choonwei/ListStudentSequenceDiagram.puml (hunk 2 above)
Comment (chwoozy) [reply docs/diagrams/choonwei/ListStudentSequenceDiagram.puml:2]: Hahaha, yes I am aware, just thought that I might need to change it next time so might as well copy first. Will remove later on if it is not needed!

**Review by JonahhGohh — APPROVED** [review]
LGTM. Not sure if we still need the d in `d:ListStudentCommand` in your UML sequence diagram. I think that d was used in DeleteCommand and it might be confusing here. I think the best representation is to remove the 'd`s all together.

**Comment by chwoozy** [comment]
> LGTM. Not sure if we still need the d in `d:ListStudentCommand` in your UML sequence diagram. I think that d was used in DeleteCommand and it might be confusing here. I think the best representation is to remove the 'd`s all together.

Okay will make changes on the next commit

**Pull request merged**